src/corecel/cont/detail/SpanImpl.hh

```diff
@@ -8,14 +8,39 @@
 #pragma once
 
 #include <cstddef>
+#include <type_traits>
 
 #include "corecel/Assert.hh"
 #include "corecel/Macros.hh"
+#include "corecel/OpaqueId.hh"
+#include "corecel/data/LdgIterator.hh"
 
 namespace celeritas
 {
 namespace detail
 {
+template<class T>
+struct SpanTraits
+{
+    using element_type = T;
+    using pointer = std::add_pointer_t<T>;
+    using const_pointer = std::add_pointer_t<T const>;
+    using iterator = pointer;
+    using const_iterator = const_pointer;
+    using reference = std::add_lvalue_reference_t<T>;
+    using const_reference = std::add_lvalue_reference_t<T const>;
+};
+template<class T>
+struct SpanTraits<LdgValue<T>>
+{
+    using element_type = typename LdgValue<T>::value_type;
+    using pointer = std::add_pointer_t<T const>;
```

Comment:
Isn't T always const here? We definitely need to document here what values of T are permissible and the fact that the `LdgValue<T>` is responsible for prohibiting (with static assert) invalid types. The `Span` class documentation also needs a description of when, where, and how `__ldg` is added to the dereferences: basically everything but `data()` calls through?

Reply:
Yes, T is always const, I specify T const again so that it's more explicit but technically not needed. `data()` is the only way to bypass `__ldg`, I'll update the doc